### PR TITLE
chore(DTFS2-7727): refactor GEF clone deal E2E tests

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:
This PR refactors the GEF clone deal tests and splits them into different tests for each type of application (eg MIN, AIN, MIA)

## Resolution :heavy_check_mark:
* Split tests into different files for E2E GEF Clone deal
* Added cypress commands to check application and facility
* Added helper to submit mock data loader deal to checker and to UKEF

### CODE REVIEW

File: e2e-tests/gef/cypress.config.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ module.exports = defineConfig({
   pageLoadTimeout: 120000,
   responseTimeout: 120000,
   numTestsKeptInMemory: 1,
-  viewportWidth: 3840,
-  viewportHeight: 2400,
+  viewportWidth: 2000,
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress.config.js:30]: I would recommend 16:9 so change that to 1920 x 1080 please.

File: e2e-tests/support/tasks.js [hunk 2]
```diff
@@ -38,6 +38,12 @@ module.exports = {
       return null;
     };
 
+    const htmlLog = (message) => {
```
Comment (abhi-markan) [inline e2e-tests/support/tasks.js:41]: Will this be removed before the PR is merged?

File: gef-ui/templates/partials/application-preview.njk [hunk 3]
```diff
@@ -32,6 +32,21 @@
     {% include "includes/application-details/supporting-information.njk" %}
   {% endif %}
 
+  <h1>CAN SUBMIT</h1>
```
Comment (abhi-markan) [inline gef-ui/templates/partials/application-preview.njk:35]: Same as above?

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: gef-ui/templates/partials/application-preview.njk (hunk 3 above)
Comment (Zainzzkk) [reply gef-ui/templates/partials/application-preview.njk]: all removed

File: e2e-tests/gef/cypress.config.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress.config.js]: done

File: e2e-tests/support/tasks.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/support/tasks.js]: removed

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-deal-validation.spec.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      });
+  });
+
+  describe('Validate the creation of a cloned deal', () => {
+    beforeEach(() => {
+      cy.saveSession();
+      cy.visit(relative(`/gef/application-details/${testDealId}`));
+    });
+
+    it('should show an error when the mandatory criteria is false', () => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-deal-validation.spec.js:29]:
```suggestion
    it('should show a not eligible error when the mandatory criteria is false', () => {
```

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-checker-deal.spec.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    before(() => {
+      cy.cloneDeal(AINdealId, clonedDealName);
+
+      /**
+       * Fetches all deals
+       * Finds the deal which is cloned by bankInternalRefName matching clonedDealName
+       * Finds the id of the facility which is in progress
+       */
+      cy.apiLogin(BANK1_MAKER1)
+        .then((tok) => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-checker-deal.spec.js:49]:
```suggestion
        .then((apiToken) => {
```

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-checker-deal.spec.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          token = tok;
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-checker-deal.spec.js:50]:
```suggestion
          token = apiToken;
```

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-ukef-deal.spec.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-ukef-deal.spec.js:1]: Please remove.

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-ukef-deal.spec.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+      before(() => {
+        cy.cloneDeal(AINdealId, clonedDealName);
+
+        /**
+         * Fetches all deals
+         * Finds the deal which is cloned by bankInternalRefName matching clonedDealName
+         * Finds the id of the facility which is in progress
+         */
+        cy.apiLogin(BANK1_MAKER1)
+          .then((tok) => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-AIN-submitted-to-ukef-deal.spec.js:54]: Same as above.

File: e2e-tests/gef/cypress/e2e/clone/clone-gef-MIA-deal.spec.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+      before(() => {
+        cy.cloneDeal(MIAdealId, clonedDealName);
+
+        /**
+         * Fetches all deals
+         * Finds the deal which is cloned by bankInternalRefName matching clonedDealName
+         * Finds the id of the facility which is in progress
+         */
+        cy.apiLogin(BANK1_MAKER1)
+          .then((tok) => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/e2e/clone/clone-gef-MIA-deal.spec.js:51]: Same as above.

File: e2e-tests/gef/cypress/fixtures/mocks/mock-deals.js [hunk 10]
```diff
@@ -213,6 +213,7 @@ export const MOCK_APPLICATION_AIN = {
 };
 
 export const MOCK_APPLICATION_MIN = {
+  version: 0,
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/fixtures/mocks/mock-deals.js:216]: Add some comments why this is set to `0` as all new deals are automatically set to `1` for FED.

File: e2e-tests/gef/cypress/support/commands/complete-manual-inclusion-supporting-info-sections.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
+import manualInclusion from '../../e2e/pages/manual-inclusion-questionnaire';
+import uploadFiles from '../../e2e/pages/upload-files';
+import relative from '../relativeURL';
+
+/**
+ * completeManualInclusionSupportingInfoSections
+ * completes the required uploads and sections to complete an MIA deal
+ * @param {String} dealId - the deal id
+ */
+const completeManualInclusionSupportingInfoSections = (dealId) => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/support/commands/complete-manual-inclusion-supporting-info-sections.js:10]: Nice!

File: e2e-tests/gef/cypress/support/commands/submit-mock-data-loader-deal-to-checker.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+ */
+const submitMockDataLoaderDealToChecker = (dealId, dealType = 'AIN') => {
+  cy.login(BANK1_MAKER1);
+
+  cy.visit(relative(`/gef/application-details/${dealId}`));
+  cy.url().should('eq', relative(`/gef/application-details/${dealId}`));
+
+  applicationDetails.automaticCoverDetailsLink().click();
+
+  if (dealType === 'AIN') {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/support/commands/submit-mock-data-loader-deal-to-checker.js:21]: Constant please for all `AIN`.

File: e2e-tests/gef/cypress/support/commands/submit-mock-data-loader-deal-to-ukef.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * submitMockDataLoaderDealToUkef
+ * completes and submits a deal from the mock data loader to UKEF
+ * calls submitMockDataLoaderDealToChecker to complete and submit the deal to the checker
+ * logs in as the checker and submits to UKEF
+ * @param {String} dealId - the deal id
+ * @param {String} dealType - if the deal if AIN or MIA
+ */
+const submitMockDataLoaderDealToUkef = (dealId, dealType = 'AIN') => {
```
Comment (abhi-markan) [inline e2e-tests/gef/cypress/support/commands/submit-mock-data-loader-deal-to-ukef.js:13]: Constant please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Nice work, only few minor comments to address please.

File: e2e-tests/gef/cypress/fixtures/mocks/mock-deals.js (hunk 10 above)
Comment (Zainzzkk) [reply e2e-tests/gef/cypress/fixtures/mocks/mock-deals.js:216]: I have removed as it is not needed anymore

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**